src/Common/LRUCache.h

```diff
@@ -64,6 +64,17 @@ class LRUCache
         setImpl(key, mapped, lock);
     }
 
+    void remove(const Key & key)
+    {
+        std::lock_guard lock(mutex);
+        auto it = cells.find(key);
+        if (it == cells.end())
+            return;
+        auto & cell = it->second;
+        current_size -= cell.size;
+        cells.erase(it);
```

Comment:
Forgot to also remove from `queue`?
btw let's remove it since it is not needed now?

Reply:
fixed